### PR TITLE
Update DHIS2 schema

### PR DESCRIPTION
**Pull request by sravfeyn**
## Product Description
Schema update to fix https://dimagi-dev.atlassian.net/browse/SC-2337. This is for DHIS2 custom integrations.

### QA Plan

NA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by sravfeyn** [comment]
Hey @kaapstorm can I get a review here?

File: corehq/motech/dhis2/schema.py [hunk 1]
```diff
@@ -217,3 +210,11 @@ def get_program_owner_schema():
         "program": id_schema,
         "trackedEntityInstance": id_schema,
     }
+
+
+def get_tracked_entity_instance_schema():
+    return {
+        "trackedEntityInstance": id_schema,
+        SchemaOptional("programOwners"): [get_program_owner_schema],
```
Comment (kaapstorm) [inline corehq/motech/dhis2/schema.py:218]:
```suggestion
        SchemaOptional("programOwners"): [get_program_owner_schema()],
```
I wonder why this didn't cause `corehq/motech/dhis2/tests/test_schema.py` to fail. (Test data is in `corehq/motech/dhis2/tests/data/tracked_entity_instance_1.json`.) Could you add "potentialDuplicate" in that tracked entity JSON doc in a way that the test fails before this change, and passes after it?

Comment (sravfeyn) [reply corehq/motech/dhis2/schema.py]: Yeah, interesting, I have added a test here and it did fail https://github.com/dimagi/commcare-hq/actions/runs/3226565110/jobs/5280236997. I have merged that branch here and it looks like it's now succeeding.

**Comment by sravfeyn** [comment]
I think the Stickler error is irrelevant `Could not fetch pull request branch.`

**Comment by sravfeyn** [comment]
@kaapstorm I have updated the test data to include this change and it [failed](https://github.com/dimagi/commcare-hq/) without the schema update and succeeds here with the schema update.

**Review by kaapstorm — APPROVED** [review]

**Pull request merged**